src/Microsoft.AspNetCore.Server.HttpSys/MessagePump.cs

```diff
@@ -250,28 +256,57 @@ private static void SetFatalResponse(RequestContext context, int status)
 
         public Task StopAsync(CancellationToken cancellationToken)
         {
-            _stopping = true;
-            // Wait for active requests to drain
-            if (_outstandingRequests > 0)
+            void RegisterCancelation()
             {
-                LogHelper.LogInfo(_logger, "Stopping, waiting for " + _outstandingRequests + " request(s) to drain.");
-
-                var waitForStop = new TaskCompletionSource<object>();
                 cancellationToken.Register(() =>
                 {
-                    LogHelper.LogInfo(_logger, "Timed out, terminating " + _outstandingRequests + " request(s).");
-                    waitForStop.TrySetResult(0);
+                    if (Interlocked.Exchange(ref _shutdownSignalCompleted, 1) == 0)
+                    {
+                        LogHelper.LogInfo(_logger, "Canceled, terminating " + _outstandingRequests + " request(s).");
+                        _shutdownSignal.TrySetResult(null);
```

Comment:
Cancelling StopAsync doesn't inherently abort requests, it just unblocks the caller so that MessagePump.Dispose() can abort all requests by calling Dispose on the underlying listener.
https://github.com/aspnet/HttpSysServer/blob/ac7d7adcb5620a2eac9c4b87286ffdb692288c72/src/Microsoft.AspNetCore.Server.HttpSys/MessagePump.cs#L310
https://github.com/aspnet/HttpSysServer/blob/33b6ed79a90bfa60eadef0b9033077ebf542cd6c/src/Microsoft.AspNetCore.Server.HttpSys/HttpSysListener.cs#L253-L260